File: 0735-asteroid-collision/0735-asteroid-collision.cpp
class Solution {
public:
    vector<int> asteroidCollision(vector<int>& asteroids) {
    stack<int>st;
    int n = asteroids.size();
    vector<int>ans;
    for(int i=0;i<n;i++){
            while(!st.empty()&&st.top()>0&&asteroids[i]<0&&
            abs(asteroids[i])>st.top()){
                st.pop();
            }
            if(st.empty()||(st.top()<0||asteroids[i]>0)) st.push(asteroids[i]);
            else if(abs(st.top())==abs(asteroids[i])){
                st.pop();
            }

        
    }  
    while(!st.empty()){
        ans.push_back(st.top());
        st.pop();
    }
    reverse(ans.begin(),ans.end());
    return ans;
    }
};

File: 0735-asteroid-collision/0735-asteroid-collision.py
class Solution(object):
    def asteroidCollision(self, asteroids):
        st=[]
        n=len(asteroids)
        ans=[]
        for i in range (n):
            while st and st[-1]>0 and asteroids[i]<0 and abs(asteroids[i])> st[-1]:
                st.pop()
            if not st or st[-1]<0 or asteroids[i]>0:
                st.append(asteroids[i])
            elif st[-1]==abs(asteroids[i]):
                st.pop()
        while st:
            ans.append(st[-1])
            st.pop() 
        ans.reverse()
        return ans